src/with-contexts.tsx

```diff
@@ -0,0 +1,44 @@
+import { BottomSheetModalProvider } from '@gorhom/bottom-sheet'
+import { DarkTheme, DefaultTheme, ThemeProvider } from '@react-navigation/native'
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
+import { Provider } from 'jotai'
+import { PostHogProvider, PostHogProviderProps } from 'posthog-react-native'
+import { ElementType } from 'react'
+import { useColorScheme } from 'react-native'
+import { GestureHandlerRootView } from 'react-native-gesture-handler'
+import { SafeAreaProvider } from 'react-native-safe-area-context'
+
+import configuration from './services/configuration'
+
+export const queryClient = new QueryClient()
+
+const getPostHogProviderProps = () => ( configuration.postHog.enabled
+  ? { apiKey: configuration.postHog.apiKey }
+  // https://github.com/PostHog/posthog-js-lite/issues/101
+  // Messes with root navigation container in tests if autocapture.captureScreens is true
+  : { apiKey: 'dummy', options: { disabled: true }, autocapture: { captureScreens: false } }
+  ) satisfies Partial<PostHogProviderProps>
+
+const withContexts = ( Component: ElementType ) => {
+  const WithContexts = () => (
+    <Provider>
+      <SafeAreaProvider>
+        <ThemeProvider value={useColorScheme() === 'dark' ? DarkTheme : DefaultTheme}>
+          <GestureHandlerRootView style={{ flex: 1 }}>
+            <BottomSheetModalProvider>
+              <QueryClientProvider client={queryClient}>
+                <PostHogProvider autocapture {...getPostHogProviderProps()}>
+                  <Component />
+                </PostHogProvider>
+              </QueryClientProvider>
+            </BottomSheetModalProvider>
+          </GestureHandlerRootView>
+        </ThemeProvider>
+      </SafeAreaProvider>
+    </Provider>
+  )
+
+  return WithContexts
+}
+
+export default withContexts
```

Comment:
we should actually start to avoid default export, they make discoverability a nightmare

Reply:
So, I've heard a lot about this and looked into it a few times. I've not landed anywhere particularly strong on the subject.

I'm not convinced they actually do limit discoverability, unless you're using something like vim lol - VS Code and IntelliJ both handle default export usage lookups + refactors + renames quite well. 

A pro of using default exports (idiomatically) can be to encourage more thought around additional exports ala single-responsibility.

Reply:
I have ran issues on my vscode time to time where it slows down due to default exports, but that’s also in a monorpeo